app/src/main/java/com/stevesoltys/seedvault/transport/backup/plugins/DocumentsStorage.kt

```diff
@@ -126,3 +133,70 @@ fun DocumentFile.deleteContents() {
 fun DocumentFile.assertRightFile(packageInfo: PackageInfo) {
     if (name != packageInfo.packageName) throw AssertionError()
 }
+
+/**
+ * Works like [DocumentFile.listFiles] except that it waits until the DocumentProvider has a result.
+ * This prevents getting an empty list even though there are children to be listed.
+ */
+@Throws(IOException::class)
+fun DocumentFile.listFilesBlocking(context: Context): ArrayList<DocumentFile> {
+    val resolver = context.contentResolver
+    val childrenUri = buildChildDocumentsUriUsingTree(uri, getDocumentId(uri))
+    val projection = arrayOf(COLUMN_DOCUMENT_ID, COLUMN_MIME_TYPE)
+    val result = ArrayList<DocumentFile>()
+
+    @SuppressLint("Recycle")  // gets closed in with(), only earlier exit when null
+    var c = resolver.query(childrenUri, projection, null, null, null)
```

Comment:
Maybe could use a more descriptive variable name here?

Reply:
Nowadays, the IDE shows you that it is a `Cursor` but sure, I changed the name.